src/Microsoft.Azure.WebJobs.Extensions.Kafka/KafkaEventDataHeaders.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.Azure.WebJobs.Extensions.Kafka
+{
+    public class KafkaEventDataHeaders : IKafkaEventDataHeaders
+    {
+        List<IKafkaEventDataHeader> headers = new List<IKafkaEventDataHeader>();
+        internal readonly bool IsReadOnly;
+
+        internal static KafkaEventDataHeaders EmptyReadOnly { get; } = new KafkaEventDataHeaders(true);
+
+        internal KafkaEventDataHeaders(bool isReadOnly = false)
+        {
+            this.IsReadOnly = isReadOnly;
+        }
+
+        internal KafkaEventDataHeaders(Confluent.Kafka.Headers headers, bool isReadOnly)
```

Comment:
Is there a scenario to use this constructor setting isReadOnly to true?

Reply:
True. Removed it.